### ISSUE TITLE
ComboboxListBoxOption: aria-selected="true" on highlight is the correct pattern based on WCAG pattern

### ISSUE THREAD

**Issue by richard-huf-bupa** [twilio-labs/paste#4381]
Hi,

It's [mentioned here](https://github.com/twilio-labs/paste/blob/f4b94cc40fdca59390cdcda5a4f994db024e1d8f/packages/paste-core/components/combobox/src/styles/ComboboxListboxOption.tsx#L104) that `aria-selected="true"` shouldn't be applied when the option is highlighted. However, looking at the WCAG authoring practices guide pattern, that is how it works. 

[WCAG APG Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-list/)

It shows how aria-selected should be [applied on an option here](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-list/#rps_label_listbox).

Thanks
